libraries/azure_policy_insights_query_result.rb

```diff
@@ -0,0 +1,98 @@
+require 'azure_backend'
+
+class AzurePolicyInsightsQueryResult < AzureResourceBase
+  name 'azure_policy_insights_query_result'
+  desc 'Lists a collection of Azure Policy Insights Query Results'
+  example <<-EXAMPLE
+    describe azure_policy_insights_query_result(policy_definition: '', resource_id: '') do
+      it { should exist }
+    end
+  EXAMPLE
+
+  def initialize(opts = {})
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+    super(opts)
+    @resource_provider = specific_resource_constraint('Microsoft.Authorization/policyDefinitions', @opts)
+    validate_parameters(required: %i(policy_definition resource_id))
```

Comment:
Hi @sathish-progress ,
While I think it's OK for the MVP, I'm not convinced we should force the provision of `policy_definitition` and `resource_id` parameters to do filtering on the API side. There are a much broader set of filters we might want to apply, and we should therefore make each one optional and build the filter dynamically.

The API can return a full set of results if we omit the API side filters, which we can then process using the filter table `.where{}` clause. That would make it quite easy to do the caching stuff we discussed on the demo.

Thoughts?

Reply:
We are doing that already in [plural resource](libraries/azure_policy_insights_query_results.rb) and here are some [examples](test/integration/verify/controls/azure_policy_insights_query_results.rb) 

This is a singular resource to send more detailed feedback to automate for individual resources..

Reply:
Ah yeah... sorry, my bad... hadn't noticed this was the singular resource.